subsys/bluetooth/host/l2cap.c

```diff
@@ -2544,14 +2546,17 @@ static void l2cap_connected(struct bt_l2cap_chan *chan)
 
 static void l2cap_disconnected(struct bt_l2cap_chan *chan)
 {
-	BT_DBG("ch %p cid 0x%04x", BT_L2CAP_LE_CHAN(chan),
-	       BT_L2CAP_LE_CHAN(chan)->rx.cid);
+	struct bt_l2cap_le_chan *ch = BT_L2CAP_LE_CHAN(chan);
+
+	BT_DBG("ch %p cid 0x%04x", ch, ch->rx.cid);
 
+#if defined(CONFIG_BT_L2CAP_DYNAMIC_CHANNEL)
```

Comment:
Looks like this entire function can be guarded. 

Basically if a function uses `BT_L2CAP_LE_CHAN` it should be guarded by `CONFIG_BT_L2CAP_DYNAMIC_CHANNEL` right?

Reply:
The necessity is not great